dataenum-processor/src/main/java/com/spotify/dataenum/processor/generator/spec/SpecTypeFactory.java

```diff
@@ -67,33 +70,45 @@ public static TypeSpec create(
       asMethods.add(valueMethods.createAsMethod(spec));
     }
 
-    TypeSpec.Builder enumBuilder =
-        TypeSpec.classBuilder(spec.outputClass())
+    TypeSpec.Builder enumBuilder  = TypeSpec.classBuilder(spec.outputClass())
             .addOriginatingElement(element)
-            .addAnnotation(
-                AnnotationSpec.builder(Generated.class)
-                    .addMember(
-                        "value", CodeBlock.of("$S", DataEnumProcessor.class.getCanonicalName()))
-                    .build())
+            .addAnnotation(getAnnotationSpec(elements))
             .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
-            .addTypeVariables(spec.typeVariables());
-
-    // add constructor with correct access
-    enumBuilder.addMethod(
-        MethodSpec.constructorBuilder()
-            .addModifiers(fromOptional(constructorAccessModifier))
-            .build());
-
-    enumBuilder.addTypes(valueTypes);
-    enumBuilder.addMethods(factoryMethods);
-    enumBuilder.addMethods(isMethods);
-    enumBuilder.addMethods(asMethods);
+            .addTypeVariables(spec.typeVariables())
+            .addMethod(
+              MethodSpec.constructorBuilder()
+                  .addModifiers(fromOptional(constructorAccessModifier))
+                  .build())
+            .addTypes(valueTypes)
+            .addMethods(factoryMethods)
+            .addMethods(isMethods)
+            .addMethods(asMethods);
 
     if (!Iterables.isEmpty(spec.outputValues())) {
-      enumBuilder.addMethod(matchMethods.createAbstractFoldVoidMethod());
-      enumBuilder.addMethod(mapMethods.createAbstractFoldMethod());
+      enumBuilder.addMethod(matchMethods.createAbstractFoldVoidMethod())
+              .addMethod(mapMethods.createAbstractFoldMethod());
     }
 
     return enumBuilder.build();
   }
+
+  private static AnnotationSpec getAnnotationSpec(Elements elements) {
+    TypeElement generatedAnnotationElement = getGeneratedAnnotationType(elements);
+
+    if (generatedAnnotationElement == null) {
```

Comment:
The behaviour in auto-value is to not include the `@Generated` annotation if not found. I sort of think this is better behaviour, but the problem is if, for some reason, it's hard for people to add a dependency that gives you the annotation. Not sure if that could be the case, but always failing in that situation is a bit harsh. It would probably be better to just make this a warning instead of failing.